stix_shifter_modules/nozomi/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,549 @@
+            comparator = f' {comparator} '
+
+        if expression.object_path == 'x-ibm-finding:finding_type':
+            formatted_value, comparator = self._handle_threat_name(formatted_value, comparator)
+
+        if mapped_field_type == 'subset':
+            mapped_fields_array = self.config_map['subset_supported_fields']
+
+        if isinstance(formatted_value, list):
+            formatted_value_list = formatted_value
+        else:
+            formatted_value_list = [formatted_value]
+
+        comparison_list = []
+
+        for value in formatted_value_list:
+            comparison_string_new = 'where '
+            for index, field_name in enumerate(mapped_fields_array):
+                if index > 0 and comparison_string_new != 'where ':
+                    comparison_string_new += ' OR '
+
+                if mapped_field_type == 'properties' and 'properties' in field_name:
+                    like_operator = self._lookup_comparison_operator(ComparisonComparators.Like)
+                    if '!' in comparator:
+                        like_operator = f'!{like_operator}'
+                    comparison_string_new += f'{field_name} {like_operator} {value}'
+                else:
+                    comparison_string = f'{field_name}{comparator}{value}'
+                    if len(comparison_string_new + comparison_string) > MAX_QUERY_LENGTH:
+                        if comparison_string_new[-4:] == ' OR ':
+                            comparison_string_new = comparison_string_new[:-4]
+                        comparison_list.append(comparison_string_new)
+                        comparison_string_new = 'where '
+                    comparison_string_new += comparison_string
+            comparison_list.append(comparison_string_new)
+
+        return comparison_list
+
+    @staticmethod
+    def check_common_timestamp(query_01, query_02):
+        """
+        Check the queries contains same timestamp
+        :param query_01: str
+        :param query_02: str
+        :return query_01_without_timestamp str, query_02_without_timestamp str, timestamp str
+        """
+        # Last 81 characters in the query string contains timestamp value
+        if query_01[-81:] == query_02[-81:]:
+            timestamp = query_02[-81:]
+            query_01_without_timestamp = query_01[:-81]
+            query_02_without_timestamp = query_02[:-81]
+            return query_01_without_timestamp, query_02_without_timestamp, timestamp
+        return query_01, query_02, None
+
+    @staticmethod
+    def _split_and_query_combine_by_or(query_01, query_02):
+        """
+        Combine AND constructed query with OR operator
+        :param query_01: str
+        :param query_02: str
+        :return query_list: list
+        """
+        query_01_split_query = query_01.split(' | ')
+        query_02_split_query = query_02.split(' | ')
+        query_list = []
+        for q1 in query_01_split_query:
```

Comment:
This gets large, very very quickly. Given the size constraints we need to be clear that combining an AND comparison with an OR comparison like (v1 AND v2) OR v3 can lead to large queries.

(v1 AND v2) OR v3 -> (v1 or v3) AND (v2 or v3)

Reply:
Due to the limitation of the data source query having |(AND operator) within brackets, we need to break it as designed. Discussed during technical review.

Reply:
Just to be clear, that it's documented that combining an AND comparison with an OR can greatly lengthen a query.

For example
(v1 AND v2) OR expression3 -> (v1 or expression3) AND (v2 or expression3)

Reply:
Added this scenario in readme.